### ISSUE TITLE
Unable to publish an aab in Internal Testing track with status completed

### ISSUE THREAD

**Issue by kgouraw** [r0adkll/upload-google-play#222]
**Describe the bug**
I used to publish an android app bundle in the Internal Testing track with my first app version 1.1
When I changed my app version to 1.2, it started throwing an error: Internal error encountered.

Expectation: It should work with status: completed without any issue.
PS: Have not changed service account JSON key

**Workflow Step Configuration**

```yml
- name: Deploy Internal
  uses: r0adkll/upload-google-play@v1.1.3
  with:
    serviceAccountJsonPlainText: ${{ secrets.PLAY_SERVICE_ACCOUNT_JSON }}
    packageName: com.example.app
    releaseFile: app/build/outputs/bundle/release/app-release.aab
    status: completed
    track: internal
    inAppUpdatePriority: 0
    changesNotSentForReview: false  
```
Error:

Creating a new Edit for this release
Validating track 'internal'
Uploading app/build/outputs/bundle/release/app-release.aab
##[debug][00653319132042375284, packageName= ****] Uploading App Bundle @
app/build/outputs/bundle/release/app-release.aab
Error: Internal error encountered.
##[debug]Node Action run completed with exit code 1
##[debug]GOOGLE_APPLICATION_CREDENTIALS='service_account.json'
##[debug]Finished: run
##[debug]Finishing: Upload Artifact to Google Play

**Step Debugging**

- [x] I have enabled [Step Debug Logging](https://docs.github.com/en/free-pro-team@latest/actions/managing-workflow-runs/enabling-debug-logging#enabling-step-debug-logging)

**Comment by boswelja** [comment]
Just double-checking, when you bump your version, you're also bumping the version code too right?

This action performs zero additional validation that would result in such an error, so it's likely this is an issue with the play console APIs

**Comment by kgouraw** [comment]
> Just double-checking, when you bump your version, you're also bumping the version code too right?
> 
> This action performs zero additional validation that would result in such an error, so it's likely this is an issue with the play console APIs

Yes, I'm bumping version code too each time with new build. Ex: Version Name:  2024.1.2 and version code 920. Next build Version: 2024.1.2 and version code 921, etc.

**Comment by kwameopareasiedu** [comment]
From [this discussion](https://github.com/fastlane/fastlane/discussions/18293) on fastlane, the Google Play Android Developer API cannot publish an internal release as completed unless the app's status is not "_Draft_".

I experimented publishing to a closed testing track first to move the app's status to "_Closed testing_", the run this action (`r0adkill/upload-google-play`) again with `track: internal` and `status: completed` and it worked.

![image](https://github.com/user-attachments/assets/0984ba9d-c21d-4b13-982f-93ed53d9ae93)

So TL'DR, first promote the manually published draft to closed testing, complete the store listing and submit for review. Then run this action...

**Comment by boswelja** [comment]
Thanks for confirming my suspicion, I'll mark this as closed 🙂

**Comment by kgouraw** [comment]
@boswelja, @kwameopareasiedu. Thank you for the replying. I tested the approach first by moving the release to the Closed testing track from the Internal track and the upload action works without any error but still it requires manual intervention. You have to remember each time that you have to promote the release before running the action.

I tried with this "changesNotSentForReview: true" and it works as per my expectation. You don't need to promote your release from Internal testing track to any other upper track.

**Comment by kwameopareasiedu** [comment]
@kgouraw Without manually moving to closed testing, the action only works with `track: internal` and `status: draft`. This is where u would require manual intervention to deploy the release.

To use `track: internal` and `status: completed` (which would publish the release automatically), that's when I found out through trial-and-error that the app's status needs to be in at least "Closed testing"...

**Comment by kgouraw** [comment]
Yes, `track: internal `and` status: draft` require manual intervention for releasing in internal track itself whereas `track: internal `and `status: completed` release in the Internal track. But both require to be promoted in the upper track before pushing again a new app version in the Internal track. Our team is big and use Internal testing track for the QA testing every day. `changesNotSentForReview: true` replaces the old the app version in the track. I hope, it's helpful.

**Comment by BobDickinson** [comment]
> But both require to be promoted in the upper track before pushing again a new app version in the Internal track. Our team is big and use Internal testing track for the QA testing every day. `changesNotSentForReview: true` doesn't send for the review and just replaces the release old the app version in the track.

I'm struggling to understand the workflow, and I'm having trouble parsing the above (particularly "pusing again a new app version").  I have a v1.0.1 app with several releases pushed to the Play Store.  In order to enable publishing of "completed" status to the "internal" track, I submitted a release to Closed testing (Alpha), went through that whole submission / approval process (took 3+ days), and then after that, I was able to push releases of v1.0.1 to internal as completed such that they became automatically available to internal testers (the goal of my whole CI/CD effort - yay!).

But now what I'm trying to understand is if I release a 1.0.2 release, am I going to have to go through that whole process again (release the first 1.0.2 release to Closed testing and get it approved) before I can push a 1.0.2 to internal as completed and have it work again?
